docs/changes.rst

```diff
@@ -21,6 +21,40 @@ Hypothesis APIs come in three flavours:
 You should generally assume that an API is internal unless you have specific
 information to the contrary.
 
+------------------
+3.7.0 - 2017-03-20
+------------------
+
+This is a feature release.
+
+New features:
+
+* Rule based stateful testing now has an @invariant decorator that specifies
+  methods that are run after init and after every step, allowing you to
+  encode properties that should be true at all times. Thanks to Tom Prince for
+  this feature.
+* The decimals strategy now supports allow\_nan and allow\_infinity flags.
+* There are significantly more strategies available for numpy, including for
+  generating arbitrary data types. Thanks to Zac Hatfield Dodds for this
+  feature.
+
+Bug fixes:
+
+* Bug fix: @composite now preserves functions docstrings.
+* The build is now reproducible and doesn't depend on the path you build it
+  from. Thanks to Chris Lamb for this feature.
+* numpy strategies for the void data type did not work correctly. Thanks to
+  Zac Hatfield Dodds for this fix. Dodds for this fix.
+
+There have also been a number of performance optimizations:
+
+* The permutations() strategy is now significantly faster to use for large
+  lists (the underlying algorithm has gone from O(n^2) to O(n))
+* Shrinking of failing test cases should have got significantly faster in
+  some circumstances where it was previously struggling for a long time.
+* Example generation now involves less indirection and so should be moderately
+  faster (probably well under 10% faster).
```

Comment:
Apparently it is weird and ambiguous, because it means the opposite of that! It was intended to convey "only a couple of percentage points faster" and failed. Will fix.